server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/CassandraDumbBlobStore.java

```diff
@@ -0,0 +1,210 @@
+    private final DataChunker dataChunker;
+    private final CassandraConfiguration configuration;
+    private final HashBlobId.Factory blobIdFactory;
+    private final BucketName defaultBucket;
+
+    @Inject
+    CassandraDumbBlobStore(CassandraDefaultBucketDAO defaultBucketDAO,
+                           CassandraBucketDAO bucketDAO,
+                           CassandraConfiguration cassandraConfiguration,
+                           HashBlobId.Factory blobIdFactory,
+                           BucketName defaultBucket) {
+        this.defaultBucketDAO = defaultBucketDAO;
+        this.bucketDAO = bucketDAO;
+        this.configuration = cassandraConfiguration;
+        this.blobIdFactory = blobIdFactory;
+        this.defaultBucket = defaultBucket;
+        this.dataChunker = new DataChunker();
+    }
+
+    @Override
+    public InputStream read(BucketName bucketName, BlobId blobId) throws ObjectStoreIOException, ObjectNotFoundException {
+        return ReactorUtils.toInputStream(readBlobParts(bucketName, blobId));
+    }
+
+    @Override
+    public Mono<byte[]> readBytes(BucketName bucketName, BlobId blobId) {
+        return readBlobParts(bucketName, blobId)
+            .collectList()
+            .map(this::byteBuffersToBytesArray);
+    }
+
+    @Override
+    public Mono<Void> save(BucketName bucketName, BlobId blobId, byte[] data) {
+        Preconditions.checkNotNull(data);
+
+        return Mono.fromCallable(() -> dataChunker.chunk(data, configuration.getBlobPartSize()))
+            .flatMap(chunks -> save(bucketName, blobId, chunks));
+    }
+
+    @Override
+    public Mono<Void> save(BucketName bucketName, BlobId blobId, InputStream inputStream) {
+        Preconditions.checkNotNull(bucketName);
+        Preconditions.checkNotNull(inputStream);
+
+        return Mono.fromCallable(() -> dataChunker.chunkStream(inputStream, configuration.getBlobPartSize()))
+            .flatMap(chunks -> save(bucketName, blobId, chunks))
+            .onErrorMap(e -> new ObjectStoreIOException("Exception occurred while saving input stream", e));
+    }
+
+    @Override
+    public Mono<Void> save(BucketName bucketName, BlobId blobId, ByteSource content) {
+        return Mono.using(content::openBufferedStream,
+            stream -> save(bucketName, blobId, stream),
+            Throwing.consumer(InputStream::close).sneakyThrow(),
+            LAZY);
+    }
+
+    private Mono<Void> save(BucketName bucketName, BlobId blobId, Flux<ByteBuffer> chunksAsFlux) {
+        return saveBlobParts(bucketName, blobId, chunksAsFlux)
+            .flatMap(numberOfChunk -> saveBlobPartReference(bucketName, blobId, numberOfChunk));
+    }
+
+    private Mono<Integer> saveBlobParts(BucketName bucketName, BlobId blobId, Flux<ByteBuffer> chunksAsFlux) {
+        return chunksAsFlux
+            .index()
+            .concatMap(pair -> writePart(bucketName, blobId, pair.getT1().intValue(), pair.getT2()).thenReturn(getChunkNum(pair)))
```

Comment:
cr/lf

Reply:
I did a bit more than that, I let you check that you are happy with the solution